repos/spack_repo/builtin/packages/cp2k/package.py

```diff
@@ -337,18 +341,21 @@ class Cp2k(MakefilePackage, CMakePackage, CudaPackage, ROCmPackage):
     # like ELPA, SCALAPACK are independent and Spack will ensure that
     # a consistent/compatible combination is pulled into the dependency graph.
     with when("+sirius"):
-        depends_on("sirius+fortran+shared")
+        depends_on("sirius+fortran+shared+scalapack")
         depends_on("sirius+cuda", when="+cuda")
         depends_on("sirius+rocm", when="+rocm")
         depends_on("sirius+openmp", when="+openmp")
         depends_on("sirius~openmp", when="~openmp")
         depends_on("sirius@7.3:", when="@9.1")
-        depends_on("sirius@7.4:7.5", when="@2023.2")
+        depends_on("sirius@7.4:", when="@2023.2")
         depends_on("sirius@7.5:", when="@2024.1:")
-        depends_on("sirius@7.6:7.7 +pugixml", when="@2024.2:")
-        depends_on("sirius@7.7: +pugixml", when="@2025.2:")
+        depends_on("sirius@7.6:+pugixml", when="@2024.2:2025.2")
+        depends_on("sirius@7.7:+pugixml", when="@2025.2:")
```

Comment:
Ranges are inclusive, so these overlap. 
```suggestion
        depends_on("sirius@7.6:+pugixml", when="@2024.2:")
        depends_on("sirius@7.7:+pugixml", when="@2025.2:")
```
?